SwaggerGen/Swagger/Type/AllOfType.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace SwaggerGen\Swagger\Type;
+
+/**
+ * allOf compound type
+ *
+ * @package    SwaggerGen
+ * @author     Bruce Weirdan <weirdan@gmail.com>
+ * @copyright  2014-2015 Martijn van der Lee
+ * @license    https://opensource.org/licenses/MIT MIT
+ */
+class AllOfType extends AbstractType
+{
+	private $allOfItems = array();
+	private $mostRecentItem;
+	protected function parseDefinition($definition)
+	{
+		$pattern = self::REGEX_START . 'allOf' . self::REGEX_CONTENT . self::REGEX_END;
+		$inlineDef = '';
+		if (preg_match($pattern, $definition, $matches)) {
+			if (isset($matches[1])) {
+				$inlineDef = $matches[1];
+			}
+		}
+		if ($inlineDef) {
+			foreach ($this->parseList($inlineDef) as $item) {
+				$this->handleCommand('item', $item);
+			}
+		}
+	}
+
+	/**
+	 * @param string $list
+	 * @return string
+	 * @todo remove duplication between this method and ObjectType::extract_property()
+	 */
+	private function getItem(&$list)
+	{
+		static $openingBraces = array('(', '[', '{', '<',);
+		static $closingBraces = array(')', ']', '}', '>');
+
+		$depth = 0;
+		$index = 0;
+		$item = '';
+
+		while ($index < strlen($list)) {
+			$c = $list{$index};
+			$index++;
+			if (in_array($c, $openingBraces)) {
```

Comment:
`in_array()` is really slow in PHP. The method in `ObjectType::extract_property()` using `strpos` should be quite a bit faster.
I also think it's a bit more readable due to keeping all the braces at the point of use, but this may be due to person taste.

Reply:
The way I use this in our project is not exactly hot path, but yeah, no reason to contribute to global warming I suppose :smiley:

Reply:
I'm generating documentation real-time in production, filtering out API calls based on authorization using the `if`/`endif` mechanism. Besides, I have a personal vendetta with `in_array()` ;)